src/components/UserSettings/VmSelect.js

```diff
@@ -0,0 +1,71 @@
+import React, { useState } from 'react'
+import { Select, SelectOption, SelectVariant } from '@patternfly/react-core'
+import { connect } from 'react-redux'
+import { withMsg } from '_/intl'
+import PropTypes from 'prop-types'
+
+const VmSelect = ({ vms, onChange, selected, msg }) => {
+  const [open, setOpen] = useState(false)
+
+  const options = vms.valueSeq().toJS()
```

Comment:
So looking at the PF4-react docs, you give the select the FULL set of possible values and let the component itself filter the list based on the input text.  That's a bit weird.

It would be nice to have a slimmed down VM fetch that only does "fetch all of the user's VMs, but only the names and ids".  Then it would be independent of the full VM details list.  And if it can be lazy fetched after the user clicks or starts typing a name, that would be nice.  The fetch could also add a search by name and require a user to type at least 1 letter before the fetch actually happens.

Reply:
> So looking at the PF4-react docs, you give the select the FULL set of possible values and let the component itself filter the list based on the input text.

Yes. This gives the best user experience when filtering the list - server side query will never keep up with typing speed.

> It would be nice to have a slimmed down VM fetch that only does "fetch all of the user's VMs, but only the names and ids". 

Oh yes:)

> And if it can be lazy fetched after the user clicks or starts typing a name, that would be nice. The fetch could also add a search by name and require a user to type at least 1 letter before the fetch actually happens.

IMHO the query is too slow for that. 

I see 2 solutions:
1. use [view more](https://www.patternfly.org/v4/components/select#view-more-with-checkboxesl) feature to let the user load next chunk of data (similar to infinite scroller) . The feature is beta and not available in our version of react-core (but PF4 is releasing fast).
2. lazy load all VMs when user enters console tab. This is not very elegant but assuming that at least 20 VMs are loaded then how many more (on avarage) require fetching? 20? 40?

Reply:
> I see 2 solutions:
> 
>   1. use [view more](https://www.patternfly.org/v4/components/select#view-more-with-checkboxes) feature to let the user load next chunk of data (similar to infinite scroller) . The feature is beta and not available in our version of react-core (but PF4 is releasing fast).
> 

That would be nice for sure, but we always need to be careful about pf4-react breaking changes.  They've happened all to frequently in the past.

>
>   2. lazy load all VMs when user enters console tab. This is not very elegant but assuming that at least 20 VMs are loaded then how many more (on avarage) require fetching? 20? 40?
>

That would be a nice thing to know, but ovirt/RHV doesn't collect statistics from installs so we can't really know for sure.  The load more option with displaying what has already been fetched (which if they hard refreshed on options would be 0 VMs) sounds like the best way to go.

Reply:
> Why not marking the the selection as invalid or clear the field?

In some case it should be possible but generally missing details does not mean that the VM is invalid. VM Portal is just not able to retrieve the details at the given moment.